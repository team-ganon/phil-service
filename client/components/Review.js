import React from 'react';
import {format, parseISO} from 'date-fns';
import style from '../CSS/review.css';

const Review = ({review}) => {
  var result = format(
    parseISO(review.date),
    'MMMM yyyy'
  );
  return (
    <div className={style.reviewOuterDiv}>
      <div className={style.topOfReview}>
        <img className={style.image} height="48" width="48" src={review.userProfile.imageUrl} />
        <div className={style.nameDate}>
          <p>{review.userProfile.name}</p>
          <p className={style.date}>{result}</p>
        </div>
      </div>
      <p className={style.reviewParagraph}>{review.body}</p>
    </div>
  );
};

export default Review;