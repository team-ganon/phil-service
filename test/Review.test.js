import React from 'react';
import renderer from 'react-test-renderer';
import Review from '../client/components/Review.js';

let reviewSample = {
  '_id': '5e718b9cce115a5271351577',
  'user': '5e718b9bce115a5271351513',
  'rental': 1,
  'body': 'Cumque pariatur quisquam quibusdam non qui qui ratione. Autem ratione rerum ut id rerum voluptatem adipisci iure accusantium. Ipsum voluptas maiores occaecati quaerat est corporis omnis perspiciatis.',
  'cleanliness': 4,
  'communication': 4,
  'value': 1,
  'accuracy': 1,
  'checkIn': 4,
  'location': 4,
  'date': '2020-03-18T02:46:52.298Z',
  '__v': 0,
  'userProfile': {
    '_id': '5e718b9bce115a5271351513',
    'name': 'David Purdy',
    'imageUrl': 'https://hackreactoramazonfrontendcapstone.s3-us-west-2.amazonaws.com/1.jpeg',
    '__v': 0
  }
};

describe('Rating Component', () => {

  it('Review SnapShot Test', () => {
    const ReviewJson = renderer.create(<Review review={reviewSample} />).toJSON();
    expect(ReviewJson).toMatchSnapshot();
  });
  
});

