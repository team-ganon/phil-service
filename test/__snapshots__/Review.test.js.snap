// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component Review SnapShot Test 1`] = `
<div
  className="reviewOuterDiv"
>
  <div
    className="topOfReview"
  >
    <img
      className="image"
      height="48"
      src="https://hackreactoramazonfrontendcapstone.s3-us-west-2.amazonaws.com/1.jpeg"
      width="48"
    />
    <div
      className="nameDate"
    >
      <p>
        David Purdy
      </p>
      <p
        className="date"
      >
        March 2020
      </p>
    </div>
  </div>
  <p
    className="reviewParagraph"
  >
    Cumque pariatur quisquam quibusdam non qui qui ratione. Autem ratione rerum ut id rerum voluptatem adipisci iure accusantium. Ipsum voluptas maiores occaecati quaerat est corporis omnis perspiciatis.
  </p>
</div>
`;
